Warm-up/DataProc.py

```diff
@@ -0,0 +1,198 @@
+        pickle.dump(np.array(nus), fh)
+
+def is_fibrillating(activity):
+
+    a = 0
+    entersfib = []
+    leavesfib = []
+    threshold = params['tissue_shape'][0]+0.05*params['tissue_shape'][0]
+	
+    def enters(c):
+    	"""" This function checks when the system first enters fibrillation from time c to time tmax """
+        for i in range(c,len(activity[0])):
+
+            if activity[0][i] >= threshold:
+                return i
+            if i+1 == len(activity[0]): 
+                raise IndexError # Simulation Terminated in Sinus Rhythm
+
+    def leaves(c):
+    	""" This function checks when the system first leaves fibrillation from time c to time tmax"""
+
+    	heart_beat_times = [j for j in range(0,len(activity[0])) if j%params['heart_rate'] == 0]
+    	b = 0
+
+    	for i in range(0, len(heart_beat_times)):  #Finds time of first heart beat after entering fibrillation and assigns b that value.
+            if heart_beat_times[i] > c:            #Where c is the time that the system entered fibrillation
+                b = heart_beat_times[i] 
+                break
+
+        for i in range(b, len(activity[0]), params['heart_rate']):
+
+            if activity[0][i] <= threshold:
+                if activity[0][i+params['heart_rate']] <= threshold:
+                    # An additional constrain on defining sinus rhythm. The average activity is less than the width of the myocardium
+                    # since the heartbeat is longer (250 > 200), i.e. there will be a period in sinus rhythm where there is 0 activity.
+                    if np.mean(activity[0][i-1:i+params['heart_rate']]) <= params['tissue_shape'][0]:  
+                        return i
+            if i+1 == len(activity[0]):
+                raise IndexError # Simulation Terminated in Fibrillatory state
+
+    while a < params['tmax']:
+        try:
+            i = enters(a)
+            entersfib.append(i)
+            a = i
+            j = leaves(a)
+            leavesfib.append(j)
+            a = j
+    	except IndexError:
+    		break
+
+    #Creating a list of 0's and 1's for each time step to note whether in fibrillatory state or not. This simplifies calculating mean_time_in_AF.
+    try:
+        is_fibrillating = [0 for i in range(0,entersfib[0])]
+    except IndexError:
+        return 0  #System never enters fibrillation
+
+    x = len(entersfib)
+    y = len(leavesfib)
+
+    for k in range(0, max(x,y)):
+        try:
+            is_fibrillating.extend([1 for i in range(entersfib[k],leavesfib[k])])
+            is_fibrillating.extend([0 for j in range(leavesfib[k],entersfib[k])])
+        except IndexError:
+            if is_fibrillating[-1] == 1:
+                is_fibrillating.extend([0 for j in range(leavesfib[k],int(params['tmax']))])
+            else:
+                is_fibrillating.extend([1 for i in range(entersfib[k],int(params['tmax']))])
+        	break
+
+    return np.mean(is_fibrillating)
+	
+def risk_curve_plot():
```

Comment:
Don't think this needs to be included here. You're just plotting something, right?